server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/VacationResponseGetMethod.scala

```diff
@@ -0,0 +1,110 @@
+import reactor.core.scala.publisher.{SFlux, SMono}
+
+object VacationResponseGetResult {
+  def found(vacationResponse: VacationResponse): VacationResponseGetResult =
+    VacationResponseGetResult(Set(vacationResponse), VacationResponseNotFound(Set.empty))
+  def notFound(vacationResponseId: UnparsedVacationResponseId): VacationResponseGetResult =
+    VacationResponseGetResult(Set.empty, VacationResponseNotFound(Set(vacationResponseId)))
+}
+
+case class VacationResponseGetResult(vacationResponses: Set[VacationResponse], notFound: VacationResponseNotFound) {
+  def merge(other: VacationResponseGetResult): VacationResponseGetResult =
+    VacationResponseGetResult(this.vacationResponses ++ other.vacationResponses, this.notFound.merge(other.notFound))
+}
+
+class VacationResponseGetMethod @Inject() (serializer: Serializer,
+                                           vacationRepository: VacationRepository,
+                                           metricFactory: MetricFactory) extends Method {
+  override val methodName: MethodName = MethodName("VacationResponse/get")
+
+  override def process(capabilities: Set[CapabilityIdentifier],
+                       invocation: Invocation,
+                       mailboxSession: MailboxSession,
+                       processingContext: ProcessingContext): Publisher[Invocation] = {
+    metricFactory.decoratePublisherWithTimerMetricLogP99(JMAP_RFC8621_PREFIX + methodName.value,
+      validate(capabilities)
+        .fold(e => SMono.just(Invocation.error(ErrorCode.UnknownMethod, invocation.methodCallId)),
+          success => asVacationResponseGetRequest(invocation.arguments)
+            .flatMap(vacationResponseGetRequest =>
+              getVacationResponse(vacationResponseGetRequest, processingContext, mailboxSession)
+                .reduce(VacationResponseGetResult(Set.empty, VacationResponseNotFound(Set.empty)), (result1: VacationResponseGetResult, result2: VacationResponseGetResult) => result1.merge(result2))
+                .map(vacationResult => VacationResponseGetResponse(
+                  accountId = vacationResponseGetRequest.accountId,
+                  state = INSTANCE,
+                  list = vacationResult.vacationResponses.toList,
+                  notFound = vacationResult.notFound))
+                .map(vacationResponseGetResponse => Invocation(
+                  methodName = methodName,
+                  arguments = Arguments(serializer.serialize(vacationResponseGetResponse).as[JsObject]),
+                  methodCallId = invocation.methodCallId)))))
+
+  }
+
+  private def validate(capabilities: Set[CapabilityIdentifier]): Either[MissingCapabilityException, Unit] =
+    if (!capabilities.contains(CapabilityIdentifier.JMAP_VACATION_RESPONSE)) {
+      Left(MissingCapabilityException(CapabilityIdentifier.JMAP_VACATION_RESPONSE))
+    } else {
+      Right()
+    }
+
+  private def asVacationResponseGetRequest(arguments: Arguments): SMono[VacationResponseGetRequest] =
```

Comment:
If the return type is an either, we would be able to write:

```
validate(capabilities)
    .flatMap(_ => asVacationResponseGetRequest(invocation.arguments))
    .fold(...)
```

I think we should rely on Either when failure is synchronous and reserve reactor usage when it is asynchronous.
```